### ISSUE TITLE
Prettier should have a better opinion on createSelector() calls

### ISSUE THREAD

**Issue by ThiefMaster** [prettier/prettier#5285]
**Prettier 1.14.3**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBzOMDKcAhgE5gAWASnGgK4A2MamAvJmCcfIfXGDBBIAKADpRMEnHkKkK1OowCSAEzQAaMZKkwAgvXqUIEALbrNkoSSPGV6zEX2GTaAJSsAfJisnbAOmNEAA5CAJbKHvaO1mgA2mEAui5iLgDcIGogEIEwIejIoKRWAO4ACqQIaMggRABuEGHpIABGJERgANbSgW0hUNjIMCS0cBm9aHAkMCWt2AHIAGYO4xkAVmgAHgBCrR3SRMZwADK9cAtLIyDdJOMkVU1ETQCe9NCNgSS9MADqYTDkyABGABMAAYMu8IOMvq1AlV3jQJjVThlOABHWghTjTIizIhnejLEDjYwhAZDC5oXrYXgARVoEHg+MJMAeP2Uf2QABYMoMiCF6FSAMImOYoKDQZEgWjjAAqD0qSEWBLgAF8VUA)
```sh
--parser babylon
--print-width 120
--tab-width 4
```

**Input:**
```jsx
const getSearchResults = createSelector(
    getSearchResultIds,
    getAllRooms,
    (roomIds, allRooms) => roomIds.map(id => allRooms[id])
);
```

**Output:**
```jsx
const getSearchResults = createSelector(getSearchResultIds, getAllRooms, (roomIds, allRooms) =>
    roomIds.map(id => allRooms[id])
);

```

**Expected behavior:**
```jsx
const getSearchResults = createSelector(
    getSearchResultIds,
    getAllRooms,
    (roomIds, allRooms) => roomIds.map(id => allRooms[id])
);
```

I understand that currently prettier applies its rules regardless of context, but in case of reselect createSelector() calls this really reduces readability. The style that can be found in pretty much all examples of reselect and projects using it is very straightforward: One input selector per line, and then the callback to combine the input.

However, due to prettier's preference of wrapping a callback argument if everything else fits in the same line (or even fitting everything in the same line), the resulting code is harder to read. It's also inconsistent if you have multiple selectors in the same file and depending on the length of their arguments they sometimes get wrapped and sometimes they don't.

One possible option for this could be to have some default overrides for certain function names, e.g. "one argument per line" for createSelector. I realize this might get close to "too much configuration" territory but it could always use a hardcoded list (or sane defaults) - also, easy-to-read/understand code should be a top priority IMHO, and that's not the case with the current formatting here.

**Comment by ThiefMaster** [comment]
@j-f1: I just saw that you added a "help wanted" label. I wouldn't mind having a look at implementing this as long as there's consensus on how it should be done (since of course I don't want to send a PR to have it rejected for "too much configuration" or something similar).

**Comment by j-f1** [comment]
I’d suggest hardcoding a list of function names (I think we already have something similar somewhere). I think we might’ve had something like this in the past, but we don’t want to cause issues with other functions that happen to be called `createSelector`.

**Comment by ThiefMaster** [comment]
If imports are tracked we could probably check if it's imported from `reselect`

**Comment by lydell** [comment]
Imports aren't tracked, and I don't think we should do that.

**Comment by ThiefMaster** [comment]
PR sent

**Comment by ikatyang** [comment]
The fix is available in [`1.15.2`](https://github.com/prettier/prettier/blob/master/CHANGELOG.md#1152).
